### ISSUE TITLE
&[Trait] should probably not be allowed

### ISSUE THREAD

**Issue by Ms2ger** [rust-lang/rust#21748]
This compiles:
```rust
trait Reader {}
fn perform(readers: &[Reader]) -> usize { 0 }
```
and this complains that there is no `len()`:
```rust
trait Reader {}
fn perform(readers: &[Reader]) -> usize { readers.len() }
```

It seems to me like both should fail to compile with an error about sticking DSTs in a slice.

**Comment by nrc** [comment]
Yeah, we should require that `T: Sized` for `[T]`, I think.

**Comment by nikomatsakis** [comment]
Note that you now get RFC 1214 warnings (i.e., this is largely fixed).

**Comment by nikomatsakis** [comment]
See e.g. http://is.gd/idE0a4

**Comment by steveklabnik** [comment]
Ahh yes, forgot about RFC 1214. Given that that will land, I think this is fine to close.
